From 85a1d5b5307604a6f081d3fdfacce26e210de5d0 Mon Sep 17 00:00:00 2001
From: Colin Watson <cjwatson@debian.org>
Date: Wed, 14 Aug 2024 15:54:15 +0100
Subject: sshd: Allow exec without absolute path in inetd mode

This restores behaviour from 9.7, and allows sshd to be executed from
xinetd again (since it only passes the basename as argv[0]).

Bug-Debian: https://bugs.debian.org/1078429
Forwarded: https://bugzilla.mindrot.org/show_bug.cgi?id=3717
Last-Update: 2024-08-22

Patch-Name: sshd-exec-inetd-relative-path.patch
---
 sshd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sshd.c b/sshd.c
index 51d5357b9..30775253b 100644
--- a/sshd.c
+++ b/sshd.c
@@ -1444,7 +1444,7 @@ main(int ac, char **av)
 			break;
 		}
 	}
-	if (!test_flag && !do_dump_cfg && !path_absolute(av[0]))
+	if (!test_flag && !inetd_flag && !do_dump_cfg && !path_absolute(av[0]))
 		fatal("sshd requires execution with an absolute path");
 
 	closefrom(STDERR_FILENO + 1 + SYSTEMD_OFFSET);
